src/main/scala/nl/knaw/dans/lib/encode/package.scala

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib
+
+import com.google.common.net.PercentEscaper
+import java.nio.file.Path
+import scala.collection.JavaConverters._
+
+package object encode {
+
+  // PercentEscaper that escapes all characters except '_' and English alphanumerical characters
+  private val bagStorePercentEscaper = new PercentEscaper("_", false)
```

Comment:
I'm not so happy with this name. `dans-scala-lib` is supposed to be 'unknowing' about anything related to EASY, Narcis, or any other of its use cases. Calling this `bagStoreXXX` pretty much makes a tight coupling it with one specific use case that happens to be used in various EASY modules. Besides, I suppose this feature will also be used for encoding paths that are being sent to other services like `easy-validate-dans-bag`.

All that to say: please call it something else 😉

Reply:
I gave it now a very generic name `percentEscaper`.  If we ever use another escape pattern for another service, we can easily modify the name to distinguish it from the other pattern, if needed.

Reply:
@rvanheest, @vesaakerman : except that this choice of safe chars is bag lib specific. I'd prefer the old name. By the same token, even more importantly, the `escapePath` function should be called `escapeBagStorePath` or something.

Reply:
@janvanmansum @vesaakerman
I would like to propose to change the signatures to:

```scala
def escapePath(implicit escaper: Escaper = percentEscaper): String
def escapeString(implicit escaper: Escaper = percentEscaper): String
```

That way we can easily swap the implementations of the `Escaper` to be used. In that case I'm fine with calling the default value for this implicit parameter `bagStorePercentEscaper` rather than `percentEscaper`.

Reply:
I also feel more for the original name, for the same reason as what @janvanmansum mentioned.

@rvanheest: it is not possible to add another parameter to the signature of a class extending `AnyVal`:
```
Since Scala 2.10, a user can create its own classes inherited from AnyVal and the compiler will generate code that won’t be using (instantiating) them directly, but some requirements must be met:

Must have only the primary constructor with exactly one val-parameter whose type is not a value class already....
```

If we in future will need another encoding for some other purpose, we can add new methods (e.g. `escapeXmlString`)  in the package, with another PercentEscaper.

@janvanmansum : I would prefer to use the simple method names `escapePath` and `escapeString`.  Even though the encoding is based on the needs of bagStore, these methods are called application wide, from 6 different modules, and this is the only encoding we have.  I would opt for this, knowing that in future we may need another encoding. That would mean at that point that we could (not necessarily) rename the present methods. Unnecessary work perhaps, but simple is beautiful...

Reply:
That was not what I meant with my suggestion, @vesaakerman. You can add the implicit parameters to the functions you define. I'm not talking about the class definitions, but the function definitions.

```diff
implicit class PathEncoding(val path: Path) extends AnyVal {

-  def escapePath: String = {
+  def escapePath(implicit escaper: Escaper = bagStorePercentEscaper): String = {
    path.asScala.map(_.toString.escapeString).mkString("/")
  }
}

implicit class StringEncoding(val s: String) extends AnyVal {

-  def escapeString: String = {
-    bagStorePercentEscaper.escape(s)
+  def escapeString(implicit escaper: Escaper = bagStorePercentEscaper): String = {
+    escaper.escape(s)
  }
}
```

This way you don't even have to define new functions when you want to use another escaper later; only when you are going to escape a new type such as `XML` (which you rightly pointed out).

Reply:
Oh yes, makes sense!  Done.